libs/dal/src/repositories/topic/topic.repository.ts

```diff
@@ -54,31 +54,40 @@ export class TopicRepository extends BaseRepository<TopicDBModel, TopicEntity, E
 
   async filterTopics(
     query: FilterQuery<TopicDBModel>,
-    pagination: { limit: number; skip: number }
-  ): Promise<TopicEntity & { subscribers: ExternalSubscriberId[] }[]> {
+    pagination: { limit: number; skip: number },
+    subscriberId?: string
+  ): Promise<{ data: TopicEntity & { subscribers: ExternalSubscriberId[] }[]; totalCount: number }> {
     const parsedQuery = { ...query };
     if (query._id) {
       parsedQuery._id = this.convertStringToObjectId(query._id);
     }
 
     parsedQuery._environmentId = this.convertStringToObjectId(query._environmentId);
     parsedQuery._organizationId = this.convertStringToObjectId(query._organizationId);
-
-    const data = await this.aggregate([
-      {
-        $match: parsedQuery,
-      },
-      lookup,
-      topicWithSubscribersProjection,
-      {
-        $skip: pagination.skip,
-      },
-      {
-        $limit: pagination.limit,
+    const subscriberFilter = subscriberId && {
+      $match: {
+        $expr: {
+          $in: [subscriberId, '$topicSubscribers.externalSubscriberId'],
+        },
       },
+    };
+    const commonAggQuery = [{ $match: parsedQuery }, lookup, ...(subscriberFilter ? [subscriberFilter] : [])];
```

Comment:
Here we are splitting the matching step in 2 different places and I do not think it is optimal. I also understand the way of fetching the data due the lookup is not a straight forward implementation.
Thus my original idea to implement this functionality was to potentially implement an endpoint:
``` GET /topic-subscribers?subscriberId=:subscriberId``` or
``` GET /topics/subscribers/:subscriberId``` to handle this functionality to simplify implementation.
Therefore playing more with the search in the subresources (topic subscribers) than in the main resource (topics).

Reply:
Thanks for the feedback @p-fernandez .

Would you prefer that we revert the entire implementation and implement a new endpoint or you would like to keep both?

Reply:
@p-fernandez gentle reminder on this

Reply:
@p-fernandez I think that if we are returning the `topics` resource here, the `subscriberId` is only a filter query to filter the topics resource. So I would stick with just leaving the query param. 
The `/topics/subscribers/:subscriberId` suggest we are returning a `subscriber` entity in the context of a topic, which is not really the case here. For example `/posts/:postId/comments` will return a list of comments related to the post vs a list of posts for a particular comment.

Wdyt?

Reply:
I agree that `/topics/subscribers/:subscriberId` could be misleading on what's being returned. Some APIs would clarify that by enabling URLs like this `/topics/all/subscribers/:subscriberId`. That will clarify what's being returned but being the topic external identifier some free choice text for our users we could hit some bug and we would need to start to forbid, thing that I am not keen to.
That's why I suggested the `GET /topic-subscribers?subscriberId=:subscriberId` option, also to not overcrowd the other option that was to implement it in the Subscribers API through `GET /subscribers/:subscriberId/topics`. But thinking in every module as it would have been a dedicated µservice I thought allow the resource `topic-subscribers` as maybe the best option.
Probably `/topics-subscribers` could be the resource proper resource.

Another worry is the usage of this endpoint. If our users would want to also return all the topic information that would need to filtered where @gitstart was doing it. But we have a problem with the API as in `TopicDto` the property subscribers is mandatory, but wouldn't make sense for this functionality, so we are between a rock and a hard place with a potential breaking change or a non performant endpoint when filtered by `suscriberId` trying to bring the relationships of the subscribers with the topics filtered.